gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/DateDiffStep.java

```diff
@@ -53,17 +54,17 @@ public DateDiffStep(final Traversal.Admin traversal, final Traversal<?, Date> da
     protected Long map(final Traverser.Admin<S> traverser) {
         final Object object = traverser.get();
 
-        if (!(object instanceof Date))
+        if (!(object instanceof OffsetDateTime))
             throw new IllegalArgumentException(
-                    String.format("DateDiff can only take Date as argument, encountered %s", object.getClass()));
+                    String.format("DateDiff can only take DateTime as argument, encountered %s", object.getClass()));
 
-        final Date otherDate = value != null ? value :
+        final OffsetDateTime otherDate = value != null ? value :
                 dateTraversal != null ? TraversalUtil.apply(traverser, dateTraversal) : null;
 
         // let's not throw exception and assume null date == 0
-        final long otherDateMs = otherDate == null ? 0 : otherDate.getTime();
+        final long otherDateMs = otherDate == null ? 0 : otherDate.toEpochSecond();
 
-        return (((Date) object).getTime() - otherDateMs) / 1000;
+        return (((OffsetDateTime) object).toEpochSecond() - otherDateMs);
```

Comment:
Did you mean to use `.toEpochMilli()`?

If you did intend to use seconds instead of milliseconds, then you should change the variable name `otherDateMs` to be `otherDateSeconds` or something.

Reply:
No this step meant to return epoch time in seconds, so I'm just directly using it.